### ISSUE TITLE
Drop on empty stack causes panic

### ISSUE THREAD

**Issue by usefulmove** [usefulmove/comp#7]
Passing the `drop` operation when the stack is empty results in a panic error with no description of the problem to the end user.

```
% comp drop
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/comp.rs:274:22
```

**Comment by Gers2017** [comment]
Hi @usefulmove there are several options to solve this, but before implementing any of them I'd like to know your thoughts on this.

```rs
fn c_drop(&mut self) {
  self.stack.pop().unwrap();
}
```
When the stack is empty, unwrapping `None` will cause a panic
```rs
fn c_drop<'a>(&mut self) -> Result<f64, &'a str> {
  match self.stack.pop() {
    Some(result) => Ok(result),
    None => Err("Custom error message")
  }
}
```
Here the `c_drop` method would return a Result enum, in case the stack isn't empty the Result would contain a number, in the opposite case it'd contain a str with your custom error message. This could be one way to solve it, not sure if it fits your needs

check this [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f1626d1f2db522d183055f75d74e50cc)

[Pop method](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.pop)
[Option enum](https://doc.rust-lang.org/stable/book/ch06-01-defining-an-enum.html?highlight=Option#the-option-enum-and-its-advantages-over-null-values)
[Error handling](https://doc.rust-lang.org/stable/book/ch09-00-error-handling.html)

**Comment by usefulmove** [comment]
@Gers2017, this is very helpful. The solution I lean toward is to check the stack (with the [Vec::is_empty()](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.is_empty) method) before popping and return from the function with a warning message to the user if the stack is empty. This seems like a condition (drop called on an empty stack) where it could be fine to continue without stopping execution. I do think it's worth a warning though to let the user know there's probably a mistake in what was passed in.

```rust
  fn c_drop(&mut self) {
    if !self.stack.is_empty() {
      self.stack.pop().unwrap();
    } else {
      println!("{}: drop called on empty stack", "warning".truecolor(255, 239, 0).bold());
    }
  }
```

I'd love to know what you think.

**Comment by usefulmove** [comment]
I looked at this again this afternoon and ended up taking a different approach. I added a generic stack depth function to check the number of elements on the stack and abort execution if the required condition is not met. I think this results in a cleaner command function implementation than my original proposal.

```rust
  fn c_drop(&mut self) {
    Interpreter::check_stack_error(self, 1, "drop");

    self.stack.pop().unwrap();
  }
```

```rust
  fn check_stack_error(&self, min_depth: usize, command: &str) {
    if self.stack.len() < min_depth {
      eprintln!("{}: {command} called without at least {min_depth} element(s) on stack", "error".truecolor(255, 0, 0).bold());
      std::process::exit(99);
    }
  }
```

**Comment by usefulmove** [comment]
Implemented with 0.20.4 on the development branch.
